src/led/libled.h

```diff
@@ -0,0 +1,492 @@
+ * @param]in]	path	Device path in include
+ * @return led_status_t	LED_STATUS_SUCCESS on sucess, else error reason.
+ * 
+ * Note: Mutually exclusive with exclude, include takes priority.
+ */
+led_status_t LED_SYM_PUBLIC led_device_include_add(struct led_ctx *ctx, const char *path);
+
+/**
+ * @brief Add a device path to an exclude list.  The library will then exclude this device
+ * from API calls.  Function can be repeated, to add to an internal list.
+ * 
+ * @param[in]	ctx	Library context 
+ * @param[in]	path	Device path to exclude 
+ * @return led_status_t LED_STATUS_SUCCESS on sucess, else error reason.
+ */
+led_status_t LED_SYM_PUBLIC led_device_exclude_add(struct led_ctx *ctx, const char *path);
+
+/**
+ * @brief Sets library blink behavior.
+ * 
+ * @param[in]	ctx	Library context 
+ * @param[in]	migration 
+ * @param[in]	init 
+ * @param[in]	all 
+ * @param[in]	raid_members 
+ * @return led_status_t LED_STATUS_SUCCESS on sucess, else error reason.
+ */
+led_status_t LED_SYM_PUBLIC led_device_blink_behavior_set(struct led_ctx *ctx, int migration,
+								int init, int all, int raid_members);
+
+/**
+ * @brief Instructs the library to scan system hardware for block devices with LED support.  This needs
+ * to be called before any other library functions can be utilized.  Can be called to update any changes
+ * of state to the hardware.
+ * 
+ * @param[in]	ctx	Library context 
+ * @return led_status_t LED_STATUS_SUCCESS on sucess, else error reason.
+ */
+led_status_t LED_SYM_PUBLIC led_scan(struct led_ctx *ctx);
+
+/**
+ * @brief Used to lookup a block device node to name used by library
+ * 
+ * @param[in] 	name			Device node to lookup
+ * @param[out] 	normalized_name		Normalized device name 
+ * @return led_status_t LED_STATUS_SUCCESS on sucess, else error reason.
+ * 
+ * Note: both parameters are expected to have a size of PATH_MAX
+ */
+led_status_t LED_SYM_PUBLIC led_device_name_lookup(const char *name, char *normalized_name);
+
+/**
+ * @brief Given a block device path, returns if it has LED hardware support via library
+ * 
+ * @param[in]	ctx	Library context 
+ * @param[in]	path	Device path 
+ * @return led_status_t LED_STATUS_SUCCESS to indicate support, else error reason.
+ */
+led_status_t LED_SYM_PUBLIC led_support(struct led_ctx *ctx, const char *path);
+
+/**
+ * @brief Turns off all LEDs for all support hardware for the system
+ * 
+ * @param[in] ctx	Library context
+ * 
+ * Note: Needs to be followed with led_flush() 
+ */
+void LED_SYM_PUBLIC led_off_all(struct led_ctx *ctx);
+
+/**
+ * @brief Retrieve the ibpi pattern for the specified device
+ * 
+ * @param[in]	ctx	Library context 
+ * @param[in]	path	Device path 
+ * @param[out]	ibpi	Current ibpi value 
+ * @return led_status_t LED_STATUS_SUCCESS on sucess, else error reason.
+ * 
+ * Note: Needs to be followed with led_flush() 
+ */
+led_status_t LED_SYM_PUBLIC led_get(struct led_ctx *ctx, const char *path, enum led_ibpi_pattern *ibpi);
```

Comment:
I think that we can drop this led_get and led_set in favor of led_get_slot and led_set_slot.
The result should be same. If you want to avoid changing it across all code- please mark those functions as backward compatibility only.
We should also export function for getting default controller (it is hidden now in led_get and led_set)
What do you think?

Reply:
Not all LED supported hardware supports get/set slot functionality, that's why this was included.

Reply:
`led_get()` - I can see that it reads current `device->ibpi` for device. That won't work in all cases. This is rather an interal ledmon value that the field we can rely on. At the begining it represents expected state for the device, the real `get` functionality was added by `ctrl_get_slot`. For me we should not export that.

`led_set()` - The subset of options for `led_set` is smaller than for `slot_set`, so simply, if we don't have `set_slot` we can fallback to old `_write` method. If `slot` property is not supported then we can throw error. For slots we can always pass `device`, it is supported. IMO library should not limit that.